src/opm/input/eclipse/Schedule/RFTConfig.cpp

```diff
@@ -174,9 +174,25 @@ void RFTConfig::update(const std::string& wname, const PLT mode)
     this->plt_state[wname] = mode;
 }
 
+void RFTConfig::update_segment(const std::string& wname, const PLT mode)
+{
+    if (mode != PLT::NO) {
+        this->seg_state.insert_or_assign(wname, mode);
+
+        return;
+    }
+
+    // mode == PLT::NO.  Remove 'wname' from list of wells for which to
+    // output segment data.
+    auto iter = this->seg_state.find(wname);
+    if (iter != this->seg_state.end()) {
+        this->seg_state.erase(iter);
+    }
```

Comment:
Any special reason not use `this->seg_state.erase(wname);` directly? Seems clearer.

Reply:
> Any special reason not use `this->seg_state.erase(wname);` directly?

Sort of.  `Erase(Iterator)` is **slightly** faster than `erase(Key)` since we don't have to perform a second lookup to locate the pertinent node.

> Seems clearer.

I don't disagree and I'll change it if you'd like me to.